File: apps/core/src/main/java/com/hnu/api/goals/GoalService.java
package com.hnu.api.goals;

import com.hnu.api.goals.operation.GoalSaveRequest;
import com.hnu.db.goals.dto.GoalDto;
import jakarta.validation.constraints.NotNull;

import java.util.List;
import java.util.UUID;

public interface GoalService {
	List<GoalDto> getGoals();

	GoalDto getGoalById(@NotNull UUID id);

	List<GoalDto> getGoalsByUserId(@NotNull UUID userId);

	UUID saveGoal(@NotNull UUID userId, GoalSaveRequest goal);

	void remove(@NotNull UUID id);
}

File: mobile-apps/hnu-app/src/app/services/goal.service.ts
import { inject, Injectable } from '@angular/core';
import { firstValueFrom } from 'rxjs';
import { GoalModel, GoalResourceService, GoalSaveRequest, SaveResult } from '@hnu-app/nu-api';

@Injectable({
  providedIn: 'root'
})
export class GoalService {
  private readonly goal = inject(GoalResourceService);

  getUserGoals(userId: string): Promise<GoalModel[]> {
    return firstValueFrom(this.goal.getUserGoals(userId));
  }

  getGoals(): Promise<GoalModel[]> {
    return firstValueFrom(this.goal.getGoals());
  }

  removeGoal(goalId: string): Promise<void> {
    return firstValueFrom(this.goal.removeGoal(goalId));
  }

  saveGoal(req: GoalSaveRequest): Promise<SaveResult> {
    return firstValueFrom(this.goal.saveGoal(req));
  }

  getGoalById(goalId: string): Promise<GoalModel> {
    return firstValueFrom(this.goal.getGoalById(goalId));
  }
}
